### ISSUE TITLE
gamepad was connected, but key doesn't work.

### ISSUE THREAD

**Issue by o8q** [360Controller/360Controller#690]
### Type of Controller
Xbox 360 controller. may be...
https://www.amazon.co.jp/gp/product/B074NWW8JS/ref=oh_aui_detailpage_o01_s00?ie=UTF8&psc=1

### Connection Method
wired

### Device Name and Info
製品ID:	0x028e
製造元ID:	0x045e  (Microsoft Corporation)
バージョン:	1.10
速度:	最高12 Mb/秒
製造元:	Xiaoji
場所ID:	0x14200000 / 2
利用可能な電流（mA）:	500
必要な電流（mA）:	500
高負荷時消費電流（mA）:	0

### Response of Controller
No response from controller

### Response in Preference Pane
no response in preference pane

### Detailed Explaination of Issue
I connect gamepad, then xbox 360 Controllers shows tell me 'Xbox 360 Wired Controller (Wired)' was connecting.
and When I push any key(X, Y, or Up...), there is no response in Controller Test Pane.

### Status kext(?)
sudo kextstat | grep com.mice
  155    0 0xffffff7f83b78000 0xc000     0xc000     com.mice.driver.Xbox360Controller (1.0.0d15) B8A01E48-7B8E-3327-8F36-321B1F346D95 <53 47 5 4 3>

### Install Log
Jun  4 14:04:24 iMac Installer[1154]: @(#)PROGRAM:Install  PROJECT:Install-1000
Jun  4 14:04:24 iMac Installer[1154]: @(#)PROGRAM:Installer  PROJECT:Installer-920
Jun  4 14:04:24 iMac Installer[1154]: Hardware: iMac13,2 @ 2.90 GHz (x 4), 8192 MB RAM
Jun  4 14:04:24 iMac Installer[1154]: Running OS Build: Mac OS X 10.13.5 (17F77)
Jun  4 14:04:24 iMac Installer[1154]: Env: __CF_USER_TEXT_ENCODING=0x1F6:0x1:0xE
Jun  4 14:04:24 iMac Installer[1154]: Env: TMPDIR=/var/folders/53/y3tbswyj6d5dd46kpr6jqkww0000gp/T/
Jun  4 14:04:24 iMac Installer[1154]: Env: HOME=/Users/o8q
Jun  4 14:04:24 iMac Installer[1154]: Env: SHELL=/bin/bash
Jun  4 14:04:24 iMac Installer[1154]: Env: SSH_AUTH_SOCK=/private/tmp/com.apple.launchd.UF9wlPykWk/Listeners
Jun  4 14:04:24 iMac Installer[1154]: Env: Apple_PubSub_Socket_Render=/private/tmp/com.apple.launchd.GUplMZIVOg/Render
Jun  4 14:04:24 iMac Installer[1154]: Env: PATH=/usr/bin:/bin:/usr/sbin:/sbin
Jun  4 14:04:24 iMac Installer[1154]: Env: LOGNAME=o8q
Jun  4 14:04:24 iMac Installer[1154]: Env: XPC_SERVICE_NAME=com.apple.xpc.launchd.oneshot.0x1000000d.Installer
Jun  4 14:04:24 iMac Installer[1154]: Env: COMMAND_MODE=unix2003
Jun  4 14:04:24 iMac Installer[1154]: Env: USER=o8q
Jun  4 14:04:24 iMac Installer[1154]: Env: XPC_FLAGS=0x0
Jun  4 14:04:26 iMac Installer[1154]: Install360Controller  Installation Log
Jun  4 14:04:26 iMac Installer[1154]: Opened from: /Volumes/360ControllerInstall/Install360Controller_0.16.8.pkg
Jun  4 14:04:26 iMac Installer[1154]: Product archive /Volumes/360ControllerInstall/Install360Controller_0.16.8.pkg trustLevel=202
Jun  4 14:04:39 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=29467=0x731b expectedThread=11523=0x2d03
Jun  4 14:04:43 iMac system_installd[517]: PackageKit: Install sandbox purging reclaimed 0 KB
Jun  4 14:04:46 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=14867=0x3a13 expectedThread=11523=0x2d03
Jun  4 14:04:46 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=14867=0x3a13 expectedThread=11523=0x2d03
Jun  4 14:04:46 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=14867=0x3a13 expectedThread=11523=0x2d03
Jun  4 14:04:46 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=14867=0x3a13 expectedThread=11523=0x2d03
Jun  4 14:04:46 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=14867=0x3a13 expectedThread=11523=0x2d03
Jun  4 14:04:46 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=14867=0x3a13 expectedThread=11523=0x2d03
Jun  4 14:05:01 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=25895=0x6527 expectedThread=11523=0x2d03
Jun  4 14:05:02 iMac Installer[1154]: InstallerStatusNotifications plugin loaded
Jun  4 14:05:13 iMac Installer[1154]: Administrator authorization granted.
Jun  4 14:05:15 iMac Installer[1154]: ================================================================================
Jun  4 14:05:15 iMac Installer[1154]: User picked Standard Install
Jun  4 14:05:15 iMac Installer[1154]: Choices selected for installation:
Jun  4 14:05:15 iMac Installer[1154]: 	インストール: "Install360Controller"
Jun  4 14:05:15 iMac Installer[1154]: 		Install360Controller_0.16.8.pkg#Install360Controller.pkg : com.mice.pkg.Xbox360controller : 0.16
Jun  4 14:05:15 iMac Installer[1154]: ================================================================================
Jun  4 14:05:15 iMac Installer[1154]: It took 0.04 seconds to summarize the package selections.
Jun  4 14:05:15 iMac deleted[397]: diskmanagement: [DMManager(PrivateMethods) clientConforms:error:]: currentThread=23567=0x5c0f expectedThread=11523=0x2d03
Jun  4 14:05:15 iMac Installer[1154]: -[IFDInstallController(Private) _buildInstallPlanReturningError:]: location = file://localhost
Jun  4 14:05:15 iMac Installer[1154]: -[IFDInstallController(Private) _buildInstallPlanReturningError:]: file://localhost/Volumes/360ControllerInstall/Install360Controller_0.16.8.pkg#Install360Controller.pkg
Jun  4 14:05:15 iMac Installer[1154]: Set authorization level to root for session
Jun  4 14:05:15 iMac Installer[1154]: Will use PK session
Jun  4 14:05:15 iMac Installer[1154]: Using authorization level of root for IFPKInstallElement
Jun  4 14:05:15 iMac Installer[1154]: Starting installation:
Jun  4 14:05:15 iMac Installer[1154]: Configuring volume "hdd"
Jun  4 14:05:15 iMac Installer[1154]: Preparing disk for local booted install.
Jun  4 14:05:15 iMac Installer[1154]: Free space on "hdd": 825.69 GB (825689894912 bytes).
Jun  4 14:05:15 iMac Installer[1154]: Create temporary directory "/var/folders/53/y3tbswyj6d5dd46kpr6jqkww0000gp/T//Install.1154vUGe1n"
Jun  4 14:05:15 iMac Installer[1154]: IFPKInstallElement (1 packages)
Jun  4 14:05:15 iMac installd[515]: PackageKit: Adding client PKInstallDaemonClient pid=1154, uid=502 (/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer)
Jun  4 14:05:16 iMac Installer[1154]: PackageKit: Enqueuing install with framework-specified quality of service (utility)
Jun  4 14:05:16 iMac installd[515]: PackageKit: ----- Begin install -----
Jun  4 14:05:16 iMac installd[515]: PackageKit: request=PKInstallRequest <1 packages, destination=/>
Jun  4 14:05:16 iMac installd[515]: PackageKit: packages=(
	    "PKLeopardPackage <id=com.mice.pkg.Xbox360controller, version=0.16, url=file://localhost/Volumes/360ControllerInstall/Install360Controller_0.16.8.pkg#Install360Controller.pkg>"
	)
Jun  4 14:05:23 iMac installd[515]: PackageKit: Extracting file://localhost/Volumes/360ControllerInstall/Install360Controller_0.16.8.pkg#Install360Controller.pkg (destination=/var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/C/PKInstallSandboxManager/6F523E6C-EE9A-49E4-A40F-5B84839F1213.activeSandbox/Root, uid=0)
Jun  4 14:05:23 iMac installd[515]: PackageKit: prevent user idle system sleep
Jun  4 14:05:23 iMac installd[515]: PackageKit: suspending Spotlight indexing
Jun  4 14:05:23 iMac installd[515]: PackageKit: suspending backupd
Jun  4 14:05:23 iMac installd[515]: PackageKit: Executing script "./preinstall" in /private/tmp/PKInstallSandbox.ikGHZ8/Scripts/com.mice.pkg.Xbox360controller.5nFHnG
Jun  4 14:05:24 iMac install_monitor[1166]: Temporarily excluding: /Applications, /Library, /System, /bin, /private, /sbin, /usr
Jun  4 14:05:24 iMac install_monitor[1166]: PackageKit: Locking applications
Jun  4 14:05:24 iMac installd[515]: PackageKit: Using trashcan path /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/T/PKInstallSandboxTrash/6F523E6C-EE9A-49E4-A40F-5B84839F1213.sandboxTrash for sandbox /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/C/PKInstallSandboxManager/6F523E6C-EE9A-49E4-A40F-5B84839F1213.activeSandbox
Jun  4 14:05:24 iMac installd[515]: PackageKit: Shoving /var/folders/zz/zyxvpxvq6csfxvn_n0000000000000/C/PKInstallSandboxManager/6F523E6C-EE9A-49E4-A40F-5B84839F1213.activeSandbox/Root (1 items) to /
Jun  4 14:05:24 iMac installd[515]: PackageKit: Executing script "./postinstall" in /private/tmp/PKInstallSandbox.ikGHZ8/Scripts/com.mice.pkg.Xbox360controller.5nFHnG
Jun  4 14:05:24 iMac installd[515]: PackageKit: kextcache -system-caches
Jun  4 14:05:28 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:05:28 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/38D01EB9-37FB-4730-828B-6303155CC4CE.kext
Jun  4 14:05:28 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klif.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/38D01EB9-37FB-4730-828B-6303155CC4CE.kext
Jun  4 14:05:28 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klif.kext) to secure location.
Jun  4 14:05:28 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:05:28 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/767E39BB-2F00-4E75-8D71-59BE27AB8B28.kext
Jun  4 14:05:28 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klnke.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/767E39BB-2F00-4E75-8D71-59BE27AB8B28.kext
Jun  4 14:05:28 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klnke.kext) to secure location.
Jun  4 14:05:30 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:05:30 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/CD809504-E34A-49FB-8CFA-5D40662EAA98.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klif.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/CD809504-E34A-49FB-8CFA-5D40662EAA98.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klif.kext) to secure location.
Jun  4 14:05:30 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:05:30 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/0A893134-5B91-462B-8985-E4534C1D1E1E.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klnke.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/0A893134-5B91-462B-8985-E4534C1D1E1E.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klnke.kext) to secure location.
Jun  4 14:05:30 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:05:30 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/5C69391B-BD26-40A0-8C6F-7B1D8DB77473.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klif.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/5C69391B-BD26-40A0-8C6F-7B1D8DB77473.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klif.kext) to secure location.
Jun  4 14:05:30 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:05:30 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/AC5BCBDC-74C1-4F3F-BAEB-563026206124.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klnke.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/AC5BCBDC-74C1-4F3F-BAEB-563026206124.kext
Jun  4 14:05:30 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klnke.kext) to secure location.
Jun  4 14:05:30 iMac installd[515]: kextcache: Kext rejected due to insecure location: <OSKext 0x7fdd5744d340 [0x7fff8b426af0]> { URL = "file:///Library/StagedExtensions/System/Library/Extensions/360Controller.kext/", ID = "com.mice.driver.Xbox360Controller" }
Jun  4 14:06:13 iMac installd[515]: PackageKit: kextcache -update-volume / -Installer
Jun  4 14:06:14 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:06:14 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/3180262A-B9D3-453C-98A3-2325A4E41DC5.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klif.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/3180262A-B9D3-453C-98A3-2325A4E41DC5.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klif.kext) to secure location.
Jun  4 14:06:14 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:06:14 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/FAAFB9EA-A804-4346-BBCE-2AD0FB58EBA2.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klnke.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/FAAFB9EA-A804-4346-BBCE-2AD0FB58EBA2.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klnke.kext) to secure location.
Jun  4 14:06:14 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:06:14 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/F7713EB2-0F39-47B9-A6D2-E13A290D44DD.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klif.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/F7713EB2-0F39-47B9-A6D2-E13A290D44DD.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klif.kext) to secure location.
Jun  4 14:06:14 iMac installd[515]: kextcache: Untrusted kexts are not allowed
Jun  4 14:06:14 iMac installd[515]: kextcache: Kext with invalid signature (-67062) denied: /Library/StagedExtensions/System/Library/Extensions/6E390124-EC6E-4C12-AEEB-1B9ACB11D8A8.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Bundle (/System/Library/Extensions/klnke.kext) failed to validate, deleting: /Library/StagedExtensions/System/Library/Extensions/6E390124-EC6E-4C12-AEEB-1B9ACB11D8A8.kext
Jun  4 14:06:14 iMac installd[515]: kextcache: Unable to stage kext (/System/Library/Extensions/klnke.kext) to secure location.
Jun  4 14:06:26 iMac installd[515]: kextcache: Kext rejected due to insecure location: <OSKext 0x7ff5be9a1d30 [0x7fff8b426af0]> { URL = "file:///Library/StagedExtensions/Library/StagedExtensions/System/Library/Extensions/360Controller.kext/", ID = "com.mice.driver.Xbox360Controller" }
Jun  4 14:06:26 iMac installd[515]: kextcache: /Library/StagedExtensions/System/Library/Extensions/360Controller.kext does not authenticate; omitting.
Jun  4 14:06:42 iMac installd[515]: kextcache: KernelCache ID: C129DA24C63CEE0DBBF19809ADB1CC14
Jun  4 14:06:42 iMac installd[515]: PackageKit: Writing receipt for com.mice.pkg.Xbox360controller to /
Jun  4 14:06:42 iMac installd[515]: PackageKit: Touched bundle /Library/Application Support/MICE/360Daemon.app
Jun  4 14:06:42 iMac installd[515]: PackageKit: ensureRecoveryBooter started.
Jun  4 14:06:42 iMac installd[515]: PackageKit: PKUpdatePrebootInstallOperation skipping disk because it is not apfs.
Jun  4 14:06:43 iMac installd[515]: Installed "Install360Controller" ()
Jun  4 14:06:43 iMac installd[515]: PackageKit: releasing Spotlight indexing
Jun  4 14:06:43 iMac install_monitor[1166]: Re-included: /Applications, /Library, /System, /bin, /private, /sbin, /usr
Jun  4 14:06:43 iMac installd[515]: PackageKit: releasing backupd
Jun  4 14:06:43 iMac installd[515]: PackageKit: allow user idle system sleep
Jun  4 14:06:43 iMac installd[515]: PackageKit: ----- End install -----
Jun  4 14:06:43 iMac installd[515]: PackageKit: 87.2s elapsed install time
Jun  4 14:06:43 iMac installd[515]: PackageKit: Running idle tasks
Jun  4 14:06:43 iMac installd[515]: PackageKit: Done with sandbox removals
Jun  4 14:06:43 iMac Installer[1154]: PackageKit: Registered bundle file:///Library/Application%20Support/MICE/360Daemon.app/ for uid 502
Jun  4 14:06:43 iMac installd[515]: PackageKit: Removing client PKInstallDaemonClient pid=1154, uid=502 (/System/Library/CoreServices/Installer.app/Contents/MacOS/Installer)
Jun  4 14:06:44 iMac Installer[1154]: Running install actions
Jun  4 14:06:44 iMac Installer[1154]: Removing temporary directory "/var/folders/53/y3tbswyj6d5dd46kpr6jqkww0000gp/T//Install.1154vUGe1n"
Jun  4 14:06:44 iMac Installer[1154]: Finalize disk "hdd"
Jun  4 14:06:44 iMac Installer[1154]: Notifying system of updated components
Jun  4 14:06:44 iMac Installer[1154]: 
Jun  4 14:06:44 iMac Installer[1154]: **** Summary Information ****
Jun  4 14:06:44 iMac Installer[1154]:   Operation      Elapsed time
Jun  4 14:06:44 iMac Installer[1154]: -----------------------------
Jun  4 14:06:44 iMac Installer[1154]:        disk      0.02 seconds
Jun  4 14:06:44 iMac Installer[1154]:      script      0.04 seconds
Jun  4 14:06:44 iMac Installer[1154]:        zero      0.07 seconds
Jun  4 14:06:44 iMac Installer[1154]:     install      88.53 seconds
Jun  4 14:06:44 iMac Installer[1154]:     -total-      88.67 seconds
Jun  4 14:06:44 iMac Installer[1154]: 
Jun  4 14:06:44 iMac Installer[1154]: IFDInstallController 13B60 state = 5
Jun  4 14:06:44 iMac Installer[1154]: Displaying 'Install Succeeded' UI.

**Comment by FranticRain** [comment]
This controller is not supported by official Xbox hardware. As such, it's support is outside the scope of this driver. We cannot and will not support this driver.

The reason you are seeing what you are seeing with the driver is because this device advertises itself to computers with the Vendor and Product ID's of a legitimate Xbox 360 controller. It violates the USB HID specification by doing this, and enters the realm of "counterfeit controllers." It would work correctly with the driver if it also imitated the initialization and packet layout of the Xbox 360 controller, but because it's designed to work with smartphones and the PS3, they certainly couldn't do this.

So basically, this device isn't compliant with the standard that this driver supports, but advertises itself like it does. Sorry. This controller is simply out of scope of this driver.

**Comment by o8q** [comment]
Thank you very much for your reply.
I am going to buy other gamepad which was supported by this driver.(with reading readme or something.)
Thank you!
